trie/child_storage.go

```diff
@@ -0,0 +1,96 @@
+package trie
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/common"
+)
+
+// ChildStorageKeyPrefix is the prefix for all child storage keys
+var ChildStorageKeyPrefix = []byte(":child_storage:")
+
+// PutChild inserts a child trie into the main trie at key :child_storage:[keyToChild]
+func (t *Trie) PutChild(keyToChild []byte, child *Trie) error {
+	childHash, err := child.Hash()
+	if err != nil {
+		return err
+	}
+
+	key := append(ChildStorageKeyPrefix, keyToChild[:]...)
+	exists, err := t.Get(key)
```

Comment:
pls rename, this is not a bool

Reply:
Really we should implement `trie.Has`. Rather than using `Get` we should isolate cases where `Has` is sufficient, so we can optimize this later (eg. maybe we don't actually need to load the value to verify)

Reply:
that would be a good idea, however I don't think trie.Has would be any more efficient that trie.Get, I don't see a way around doing the lookup right now. but it would be cool, I will think about it

Reply:
I've modified the trie.insert function to update the value if we are re-inserting into the same key, got rid of this part of the code